File: src/backend/internal/domain/project/phase6.go
package project

import "time"

type FunctionCoverage struct {
	Name    string  `json:"name"`
	Source  string  `json:"source"`
	Percent float64 `json:"percent"`
}

type PackageCoverage struct {
	Package string  `json:"package"`
	Percent float64 `json:"percent"`
}

type Phase6CoverageReport struct {
	GeneratedAt      time.Time          `json:"generated_at"`
	ThresholdPercent float64            `json:"threshold_percent"`
	TotalPercent     float64            `json:"total_percent"`
	Packages         []PackageCoverage  `json:"packages"`
	Functions        []FunctionCoverage `json:"functions"`
	RawGoTestOutput  string             `json:"raw_go_test_output,omitempty"`
}

type TestFailureKind string

const (
	TestFailureNone               TestFailureKind = "NONE"
	TestFailureTestImplementation TestFailureKind = "TEST_IMPLEMENTATION"
	TestFailureProjectBug         TestFailureKind = "PROJECT_BUG"
)

type Phase6ValidationResult struct {
	GoTestPassed       bool            `json:"go_test_passed"`
	GoRacePassed       bool            `json:"go_race_passed"`
	FrontendTestPassed bool            `json:"frontend_test_passed"`
	FailureKind        TestFailureKind `json:"failure_kind"`
	Details            string          `json:"details,omitempty"`
}

File: src/frontend/src/types/phase6.ts
export interface PackageCoverage {
  package: string;
  percent: number;
}

export interface FunctionCoverage {
  name: string;
  source: string;
  percent: number;
}

export interface Phase6CoverageReport {
  generated_at: string;
  threshold_percent: number;
  total_percent: number;
  packages: PackageCoverage[];
  functions: FunctionCoverage[];
  raw_go_test_output?: string;
}

export interface Phase6AnalyzeCoverageResponse {
  below_threshold: boolean;
  report: Phase6CoverageReport;
}

export type TestFailureKind = "NONE" | "TEST_IMPLEMENTATION" | "PROJECT_BUG";

export interface Phase6ValidationResult {
  go_test_passed: boolean;
  go_race_passed: boolean;
  frontend_test_passed: boolean;
  failure_kind: TestFailureKind;
  details?: string;
}
